machinery/core_types.go

```diff
@@ -0,0 +1,91 @@
+package machinery
+
+import (
+	core "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	gwapiv1 "sigs.k8s.io/gateway-api/apis/v1"
+)
+
+var (
+	ServiceGroupKind     = core.SchemeGroupVersion.WithKind("Service").GroupKind()
```

Comment:
Maybe add it for Namespace too?

```suggestion
	NamespaceGroupKind   = core.SchemeGroupVersion.WithKind("Namespace").GroupKind()
	ServiceGroupKind     = core.SchemeGroupVersion.WithKind("Service").GroupKind()
```